benchmark/scripts/run_smoke_bench

```diff
@@ -141,26 +139,23 @@ def test_opt_levels(args):
 
     if output_file:
         if changes:
-            output_file.write(get_info_text())
+            output_file.write(get_info_text(args.threshold))
         else:
             output_file.write("### No performance and code size changes")
         output_file.close()
     return 0
 
 
 def measure(driver, tests, i):
-    """Log and measure samples of the tests with the given driver.
-
-    Collect increasing number of samples, depending on the iteration.
-    """
-    num_samples = min(i + 3, 10)
-    msg = '    Iteration {0} for {1}: num samples = {2}, '.format(
-        i, driver.args.tests, num_samples)
+    """Log and measure samples of the tests with the given driver."""
+    msg = '    Iteration {0} for {1}: '.format(i, driver.args.tests)
     msg += ('running all tests' if driver.all_tests == tests else
             're-testing {0} tests'.format(len(tests)))
     log(msg)
     driver.tests = tests
-    return driver.run(num_samples=num_samples, sample_time=0.0025)
+    return driver.run(
+        num_iters=1, min_samples=10, sample_time=0.25, quantile=20,
```

Comment:
It is not a typo. I've intended to gather samples for a quarter of a second to ensure benchmarks have reached steady state. From PR description:
> This PR adjusts the measurement method (in both the Benchmark_Driver as well as run_smoke_bench) to sample each benchmark for 250 ms and gather at minimum 10 samples. The run_smoke_bench now gathers up to 10 independent runs for the suspected changes. Ten independent runs with larger sample counts are enough to get statistically sound results, so it no longer spins in a measurement loop waiting for 10 no-change runs in a row.

Of course it makes the measurement take longer than before, because we are gathering more samples. The 2500 μs you've used originally were a reasonable trade-off at time when measuring the whole benchmark suite took much longer. But the proper baseline to compare this is before all the **legacy factor** refactoring. These have massively shortened the time to run the whole SBS. The point here is to use the time we've won and invest it into better quality of measurement: more samples. This way we can rely on better statistical methods in our analysis.

Here are my local timings per one full run of SBS with in various cases:

options | `O` | `Osize` | `Onone`
--|--|--|--
`--sample-time=0.0025 --num-samples=3` | 0:10 | 0:12 | 1:31
`--sample-time=0.0025 --num-samples=10` | 0:31 | 0:39 | 4:43
`--sample-time=0.25 --min-samples=10`<br>` --num-iters=1` | 1:31 | 2:07 | 6:12

There were 758 benchmarks in the pre-commit category of the SBS when I measured this locally, which — at 0.25 s per benchmark — should nominally take **3:10** for one full run of SBS.  Comparing this to the last line with my proposed measurement parameters, `O` and `Osize` take less time, because measurements are capped at 200 samples and end earlier for fast benchmarks. The `Onone` takes more than that because on my slow machine it takes longer to gather at least 10 samples per benchmark (that's also visible in the `--num-samples=10` case, which be the worst case scenario per iteration in our current system).

Your modern machines are much faster. Looking at the [console of the benchmark run](https://ci.swift.org/job/swift-PR-osx-perf/1156/console) in this PR and comparing it to [ numbers from another benchmark run](https://ci.swift.org/job/swift-PR-osx-smoke-perf/924/console) of the current `run_smoke_bench` from PR 26630 yields following times for single 0 iteration of SBS:

options | `O` | `Osize` | `Onone`
--|--|--|--
`--sample-time=0.0025 --num-samples=3` | 0:06 | 0:07 | 0:31
`--sample-time=0.25 --min-samples=10`<br>` --num-iters=1` | 0:34 | 0:39 | 2:21

The total time to run smoke benchmark was 3:19 with current method and 8:09 with the one in this PR, a slowdown of approximately 2.5. Note that the time from initial git pull to finished compilation on this PR was 19:23.

Reply:
Note that going back to capping the iterations at 10 (after addressing the “instabilities”), might slightly improve the total runtime of this method, to maybe somewhere around 2x slowdown.

Reply:
I will also look at running benchmarks in parallel with this method, which will speed this back up again, depending on the number of cores on the machine.